app/javascript/lib/metrics-api.js

```diff
@@ -149,6 +149,22 @@ function getClassListAsArray(target) {
   return 'classList' in target? Array.from(target.classList) : []
 }
 
+/** Log a tab click if click was on a tab otherwise return false */
```

Comment:
changing this function signature to `updatePropertiesIfTab(target, props, eventName)` and having it return eventName if not a tab click, and `click:tab` if it is a tab click will let you avoid the funky `alreadyLoggedTabClick` ternary expressions you have below.

Reply:
To clarify, I understand this as suggesting the function would return `eventName` _and_ the updated log props.

Reply:
it could, but I would just as soon avoid multiple return values if we can help it. it can update the properties in-place, which is why the method is called 'update...' A more purely functional approach might, as Eric said, be a method called getEventPropsWithTabsApplied() that returns both and leaves the passed-in props unchanged. Both are totally reasonable approaches